src/trio/_path.py

```diff
@@ -32,219 +21,97 @@
         OpenBinaryModeWriting,
         OpenTextMode,
     )
-    from typing_extensions import Concatenate, Literal, ParamSpec, TypeAlias
-
-    from trio._file_io import AsyncIOWrapper as _AsyncIOWrapper
+    from typing_extensions import Concatenate, Literal, ParamSpec
 
     P = ParamSpec("P")
 
-T = TypeVar("T")
-StrPath: TypeAlias = Union[str, "os.PathLike[str]"]  # Only subscriptable in 3.9+
+    Self = TypeVar("Self", bound="Path")
+    T = TypeVar("T")
 
 
-# re-wrap return value from methods that return new instances of pathlib.Path
-def rewrap_path(value: T) -> T | Path:
-    if isinstance(value, pathlib.Path):
-        return Path(value)
-    else:
-        return value
+def _wraps_async(
+    wrapped: Callable[..., Any]
+) -> Callable[[Callable[P, T]], Callable[P, Awaitable[T]]]:
+    def decorator(fn: Callable[P, T]) -> Callable[P, Awaitable[T]]:
+        async def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
+            return await run_sync(partial(fn, *args, **kwargs))
 
+        update_wrapper(wrapper, wrapped)
+        wrapper.__doc__ = (
+            f"Like :meth:`~{wrapped.__module__}.{wrapped.__qualname__}`, but async."
+        )
```

Comment:
Would it be possible to include the original documentation after this header? I've personally always found it frustrating when I'm looking into what trio path objects can do and all the documentation just says "hey go look at this other thing" instead of it just being right there.

Reply:
I'd prefer that too! I was just following convention! I wonder if anyone else has any opinions.

Reply:
I suspect the current convention is out of laziness/complexity, rather than it being the optimal thing :tm:. If you can copy the old one and prepend that text, or something similar, I think it would be much superior.

Reply:
I'll do that this evening. A good call! I actually had it set up this originally, then reigned in my ambitions and followed conventions. 😅